go-controller/hybrid-overlay/pkg/controller/node_linux.go

```diff
@@ -67,11 +70,58 @@ func (n *NodeController) addOrUpdatePod(pod *kapi.Pod) error {
 	if err != nil {
 		return fmt.Errorf("failed to add flows for pod %s/%s: %v", pod.Namespace, pod.Name, err)
 	}
+	subnet, err := getLocalNodeSubnet(n.nodeName)
+	if err != nil {
+		return err
+	}
+	namespace, err := n.kube.KClient.CoreV1().Namespaces().Get(pod.Namespace, metav1.GetOptions{})
+	if err != nil {
+		klog.V(5).Infof("unable to get namespace %s annotations", pod.Namespace)
+	}
+	if namespace == nil {
+		return nil
+	}
+	podExternalGW, ok := namespace.GetAnnotations()[hotypes.HybridOverlayExternalGw]
+	if ok {
+		_, _, err = util.RunOVSOfctl("add-flow", extBridgeName,
+			fmt.Sprintf("table=0, cookie=0x%s, priority=100, ip, nw_src=%s, actions=load:%s->NXM_NX_TUN_ID[0..31],set_field:%s->tun_dst,output:%s", cookie, podIP.IP, strconv.Itoa(hotypes.HybridOverlayExternalGwVNI), podExternalGW, extVXLANName))
+		if err != nil {
+			return fmt.Errorf("failed to add flows for pod %s/%s: %v", pod.Namespace, pod.Name, err)
+		}
+		firstIP, secondIP := util.GetNodeWellKnownAddresses(subnet)
```

Comment:
@rcarrillocruz maybe we should just update GetNodeWellKnownAddresses() to return the 3rd IP too.

Reply:
or have three separate functions? Many callers don't want both firstIP and secondIP anyway. (eg, the code here wouldn't be using secondIP if the function had returned thirdIP directly).